### ISSUE TITLE
Returning the maximum value of a level

### ISSUE THREAD

**Issue by jbe456** [atoti/atoti#38]
## Context

This code sample is only useful to understand the use case of what I wanted to achieve:
```python
import atoti as tt
import numpy as np
import pandas as pd

session = tt.create_session()

drivers_df = pd.read_csv(
    "https://data.atoti.io/notebooks/formula-one/drivers.csv", encoding="latin-1"
)
drivers_df.rename(columns={"url": "driver_url"}, inplace=True)
drivers_df.rename(columns={"nationality": "driver_nationality"}, inplace=True)

drivers_store = session.read_pandas(
    drivers_df, keys=["driverId"], store_name="F1 drivers"
)

races_df = pd.read_csv(
    "https://data.atoti.io/notebooks/formula-one/races.csv", encoding="latin-1"
)
races_df.rename(columns={"url": "race_url"}, inplace=True)
races_df.rename(columns={"name": "race_name"}, inplace=True)

races_store = session.read_pandas(
    races_df, keys=["raceId"], store_name="F1 races"
)

constructors_df = pd.read_csv(
    "https://data.atoti.io/notebooks/formula-one/constructors.csv", encoding="latin-1"
)
constructors_df.rename(columns={"url": "constructor_url"}, inplace=True)
constructors_df.rename(columns={"name": "constructor_name"}, inplace=True)
constructors_df.rename(columns={"nationality": "constructor_nationality"}, inplace=True)

constructors_store = session.read_pandas(
    constructors_df,
    keys=["constructorId"],
    store_name="F1 constructors",
)

resultsTypes = {
    "points": tt.types.DOUBLE,
}

results_store = session.read_csv(
    "s3://data.atoti.io/notebooks/formula-one/results.csv",
    keys=["resultId"],
    store_name="F1 results",
    types=resultsTypes,
)

results_store.join(drivers_store, mapping={"driverId": "driverId"})
results_store.join(races_store, mapping={"raceId": "raceId"})
results_store.join(constructors_store, mapping={"constructorId": "constructorId"})

f1cube = session.create_cube(results_store, "F1Cube")
session.load_all_data()

lvl = f1cube.levels
m = f1cube.measures
h = f1cube.hierarchies
```

## Use Case

The use case is to return the maximum points among all drivers given any cube location. Think of it as the value used to determine `agg.max_member`.

Therefore I wanted to aggregate the maximum driver's `points.SUM` always across all drivers: i.e. slicing by driver but also ignoring any filter on the driver level.

I thought about using `scope.siblings` to achieve that [with a level as an argument](https://github.com/atoti/atoti/issues/39).

```python
m["Champion Points"] = tt.agg.max(
    m["points.SUM"], scope=tt.scope.siblings(lvl["driverRef"])
)
```

While it works when the level driver is explicitly expressed, it does not otherwise: indeed since it takes a hierarchy as an input, if the level driver is not explicitly expressed, it will return points.SUM for AllMember on the `driverRef` hierarchy.

I also thought at some point we could introduce something like:
```python
m["Champion Points"] = tt.agg.max(
    m["points.SUM"], scope=tt.scope.origin_ignore_filters(lvl["driverRef"])
)
```

The "workaround" was to do the following:
```python
driver_points_max = tt.agg.max(m["points.SUM"], scope=tt.scope.origin(lvl["driverRef"]))
m["Champion Points"] = tt.parent_value(
    driver_points_max, on=[h["driverRef"]], top_value=driver_points_max
)
```

I was able to achieve what I wanted so this is more for documentation purpose and let you decide if you want to improve the API or not.

If you decide that the above syntax is the right one, could you document it in the tutorial next to parent value? It seems a legit use case for `parent_value`.

## Environment

- Atoti: 0.4.0.20200512194154
- Python: 3.8.2
- Operating System: linux

**Comment by fabiencelier** [comment]
I don't think what you described is a workaound, you should use the `max` with `tt.scope.origin` followed by a `parent_value`.

There is [total](https://github.com/activeviam/atoti/issues/1329) coming soon that will let you write it in a similar way but slightly more intuitive:

```python
driver_points_max = tt.agg.max(m["points.SUM"], scope=tt.scope.origin(lvl["driverRef"]))
m["Champion Points"] = tt.total(driver_points_max, on=h["driverRef"])
```

total is similar to parent value but easier: it take the value corresponding to all the members ignoring the filters.

**Comment by rcolle** [comment]
I'm closing this since `total` was released with 0.4.0.
